www/main/common/store/user-settings/create-user-settings-state.js

```diff
@@ -16,6 +16,45 @@ const createUserSettingsState = (settingsSchema, savedSettings, userConfigPath)
       userSettingsState[stateVarName] = savedSettings[settingKey];
     }
 
+    // Defines an action to reset all font size settings to their initial values.
+    userSettingsState.resetFontSizes = action((state) => {
```

Comment:
We shouldn't have to rewrite all the logic here. This is automatically generated if we add a new setting in user-settings.json file.

So the flow would be
- Add a setting in user-settings.json in category `font-sizes` In file user-settings.json
```json
 "font-sizes": {
      "title": "FONT_SIZES",
      "type": "range",
      "max": 20,
      "min": 1,
      "step": 1,
      "settings": [
        "gurbani-font-size",
        "content1-font-size",
        "content2-font-size",
        "content3-font-size",
        "announcements-font-size",
        <add-new -setting-here>
      ]
    },
```

- In same file, inside settings object, add a new setting for reset button, you can add a new type for this, let's call it reset-button, and another property named resetSettings, that will give an array of settings it will reset.
```
"settings": {
    "<new-setting-here>": {
      "title": "SOME NAME HERE",
      "type": "reset-button"
      "resetSettings": ['gurbani-font-size', 'content1-font-size', 'content2-font-size', 'content3-font-size', 'announcement-font-size']
    },
```

- In Setting.jsx, add a new type inside switch statement for this, where you add a button and reset these settings using onclick attribute.

To understand more about how its working, read through this [documentation](https://github.com/KhalisFoundation/sttm-desktop/wiki/Settings-Doc-Part-1:-The-structure-of-user-settings.json-(feat-basics-of-global-state-management)#structure-of-user-settingsjson) and we can discuss if you have any doubts.